### PR TITLE
Microsoft.Security new version - 2024-02-01

### PR DESCRIPTION
**Pull request by aviramyi**
# ARM (Control Plane) API Specification Update Pull Request 

> Tip: overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

<details>
<summary> Click here to see the details of Step 1 </summary>

### Breaking changes review (Diagram Step 1)

If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
- The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
- You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

</details>

<details> 
<summary> Click here to see the details of Step 2 </summary>

### ARM API changes review (Diagram Step 2)

- If this PR is in purview of ARM review then automation will add the `ARMReview` label.
- If you want to force ARM review, add the label yourself.
- Proceed according to the diagram at the top of this comment.

</details>

<details>
<summary> Click here to see the diagram footnotes</summary>

### Diagram footnotes

[1] [ARM review queue] (for **merge** queues, see [2])  
The PRs are processed by time opened, ascending. Your PR may show up on 2nd or later page. 
If you addressed Step 1 from the diagram and your PR is not showing up in the queue, ensure the label `ARMChangesRequested` 
is removed from your PR. This should cause the label `WaitForARMFeedback` to be added.
[2] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])

If you need further help with anything, see `Getting help` section below.

</details>

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [x] New API version. (If API spec is not defined in TypeSpec, the PR should have been generated using [OpenAPI Hub](https://aka.ms/openapihub)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the Diagram Step 2, "ARM API changes Review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/azsdk/pr-suppressions) 
to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://github.com/search?q=org%3AAzure+is%3Apr+is%3Aopen+label%3AWaitForARMFeedback+-label%3AIDCDevDiv++draft%3Afalse+sort%3Acreated-asc+&type=pullrequests
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

**Review by kerend — APPROVED** [review]

File: specification/security/resource-manager/Microsoft.Security/preview/2024-02-01-preview/assessmentMetadata.json
Comment (raosuhas) [inline specification/security/resource-manager/Microsoft.Security/preview/2024-02-01-preview/assessmentMetadata.json:4]: Can you please add the prevvious versions of these 2 files in the first commit so that we canreview the diff across the versions ?

**Comment by raosuhas** [comment]
> Hi @aviramyi! For review efficiency consideration, when creating a new API version, it is required to place API specs of the base version in the first commit, and push new version updates into successive commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

This is flagged as part of the automated tests as well here

File: specification/security/resource-manager/Microsoft.Security/preview/2024-02-01-preview/assessmentMetadata.json
Comment (aviramyi) [reply specification/security/resource-manager/Microsoft.Security/preview/2024-02-01-preview/assessmentMetadata.json:4]: Hi, I am not sure I follow - both `assessments.json` and `assessmentMetadata.json` are included in the first commit as a copy from the 202without any changes, with their previous version date (`2021-06-01`). Appreciate your help

Comment (raosuhas) [reply specification/security/resource-manager/Microsoft.Security/preview/2024-02-01-preview/assessmentMetadata.json:4]: you need to paste it in the same folder otherwise , we cannot do a diff. 

Your first commit is this one : https://github.com/Azure/azure-rest-api-specs/pull/27389/commits/e4f6045fe7851b06e4d251a804d149428fcb7cb7 and you created the files under the "2023-04-01-preview" folder 

THe current review is for the 2024-02-01-preview folder. The first commit should be with the previous version in the same folder , this enables reviewers to diff the changes across commits..

For now you can just create a new commit in the same folder and then overwrite it with another commit and let me know th commit id so that i know which to diff from.,

File: specification/security/resource-manager/Microsoft.Security/preview/2024-02-01-preview/assessments.json
Comment (visingla-ms) [inline specification/security/resource-manager/Microsoft.Security/preview/2024-02-01-preview/assessments.json:517]: The id, type etc. are some of the standard ARM property names, which ARM as well as other RPs assume to be part of an ARM resource. So, unless it has to be this name of the property for some reason, please use some different name e.g., itemId or referenceItemId here.

File: specification/security/resource-manager/readme.md
Comment (kazrael2119) [inline specification/security/resource-manager/readme.md:417]: why you remove these resource types?

Comment (aviramyi) [reply specification/security/resource-manager/readme.md:417]: These were all duplicates - you can see that they are still available under the `package-composite-v3` tag.

File: specification/security/resource-manager/Microsoft.Security/preview/2024-02-01-preview/assessments.json
Comment (aviramyi) [reply specification/security/resource-manager/Microsoft.Security/preview/2024-02-01-preview/assessments.json:517]: Hi, thank you for the comment 😃
The `risk` object (which is where the mentioned properties appear) will never be an independent ARM resource (given how the product and properties are designed). Furthermore, it will always be a property inside an ARM resource, which means that the "type" and "id" properties won't collide with the ARM mandatory properties.

The mentioned names are correlated to an external database that is external to customer, so for the customer benefit we would like to keep these names for easier correlation.

**Comment by kazrael2119** [comment]
please fix duplicate schemas to make sdk generation work, more detail please see https://github.com/Azure/sdk-release-request/issues/4762#issuecomment-1827286038 and https://github.com/Azure/azure-rest-api-specs/pull/27389/checks?check_run_id=21141086420

**Comment by aviramyi** [comment]
> please fix duplicate schemas to make sdk generation work, more details please see [Azure/sdk-release-request#4762 (comment)](https://github.com/Azure/sdk-release-request/issues/4762#issuecomment-1827286038) and https://github.com/Azure/azure-rest-api-specs/pull/27389/checks?check_run_id=21141086420

Hi @kazrael2119,
I reviewed the attached link.
Unfortunately, each of the mentioned issues is under a different team that share Microsoft.Security RP.

Given that information, and the fact that this spans across teams, I would really appreciate if we can bypass this validation, as **my PR didn't add any errors that weren't there before**.
[Here is a PR ](https://github.com/Azure/azure-rest-api-specs/pull/25655/checks)that was approved and merged with these issues.

Appreciate your assistance.

**Comment by kazrael2119** [comment]
> > please fix duplicate schemas to make sdk generation work, more details please see [Azure/sdk-release-request#4762 (comment)](https://github.com/Azure/sdk-release-request/issues/4762#issuecomment-1827286038) and https://github.com/Azure/azure-rest-api-specs/pull/27389/checks?check_run_id=21141086420
> 
> Hi @kazrael2119, I reviewed the attached links and errors. Unfortunately, each of the mentioned issues is under a different team that share Microsoft.Security RP.
> 
> Given that information, and the fact that this spans across teams, I would really appreciate if we can bypass this validation, as **my PR didn't add any errors that weren't there before**. [Here is a PR ](https://github.com/Azure/azure-rest-api-specs/pull/25655/checks)that was approved and merged with these issues.
> 
> Appreciate your assistance.

could you help check this duplicate?
image


and could you ask pricing team to fix the duplicate operations?
```
Duplicates have those paths:
  - get /subscriptions/{subscriptionId}/providers/Microsoft.Security/pricings
  - get /{scopeId}/providers/Microsoft.Security/pricings
error   | DuplicateOperation | Duplicate Operation 'Pricings' > 'Get' detected(This is most likely due to 2 operation using the same 'operationId' or 'tags'). Duplicates have those paths:
  - get /subscriptions/{subscriptionId}/providers/Microsoft.Security/pricings/{pricingName}
  - get /{scopeId}/providers/Microsoft.Security/pricings/{pricingName}
error   | DuplicateOperation | Duplicate Operation 'Pricings' > 'Update' detected(This is most likely due to 2 operation using the same 'operationId' or 'tags'). Duplicates have those paths:
  - put /subscriptions/{subscriptionId}/providers/Microsoft.Security/pricings/{pricingName}
  - put /{scopeId}/providers/Microsoft.Security/pricings/{pricingName}
```
this caused the sdk generation failed, and we can't approve any sdk breaking if the sdk generation is failed

**Comment by kazrael2119** [comment]
> > > please fix duplicate schemas to make sdk generation work, more details please see [Azure/sdk-release-request#4762 (comment)](https://github.com/Azure/sdk-release-request/issues/4762#issuecomment-1827286038) and https://github.com/Azure/azure-rest-api-specs/pull/27389/checks?check_run_id=21141086420
> > 
> > 
> > Hi @kazrael2119, I reviewed the attached links and errors. Unfortunately, each of the mentioned issues is under a different team that share Microsoft.Security RP.
> > Given that information, and the fact that this spans across teams, I would really appreciate if we can bypass this validation, as **my PR didn't add any errors that weren't there before**. [Here is a PR ](https://github.com/Azure/azure-rest-api-specs/pull/25655/checks)that was approved and merged with these issues.
> > Appreciate your assistance.
> 
> could you help check this duplicate? image
> 
> and sdk generation failure is caused by the following error:
> 
> ```
> error   | DuplicateOperation | Duplicate Operation 'Pricings' > 'List' detected(This is most likely due to 2 operation using the same 'operationId' or 'tags'). Duplicates have those paths:
>   - get /subscriptions/{subscriptionId}/providers/Microsoft.Security/pricings
>   - get /{scopeId}/providers/Microsoft.Security/pricings
> error   | DuplicateOperation | Duplicate Operation 'Pricings' > 'Get' detected(This is most likely due to 2 operation using the same 'operationId' or 'tags'). Duplicates have those paths:
>   - get /subscriptions/{subscriptionId}/providers/Microsoft.Security/pricings/{pricingName}
>   - get /{scopeId}/providers/Microsoft.Security/pricings/{pricingName}
> error   | DuplicateOperation | Duplicate Operation 'Pricings' > 'Update' detected(This is most likely due to 2 operation using the same 'operationId' or 'tags'). Duplicates have those paths:
>   - put /subscriptions/{subscriptionId}/providers/Microsoft.Security/pricings/{pricingName}
>   - put /{scopeId}/providers/Microsoft.Security/pricings/{pricingName}
> ```
> 
> I think after this pr #27497 merged into main, these will be fixed

**Comment by JeffreyRichter** [comment]
Yes, Azure's policy is that a "service" (RP) is a set of operation endpoints that version uniformly together. And, only after this is done properly can customers use the documentation correctly and SDK packages can be made. In addition, we cannot ship any SDK package where some api-versions are preview and some are GA. The api-versions must be unified before this PR can be approved.

**Pull request closed without merging**
